reactive/apache_bigtop_base.py

```diff
@@ -19,3 +32,20 @@ def fetch_bigtop():
         hookenv.status_set('waiting',
                            'Unable to fetch Bigtop: checksum error'
                            ' (will retry)')
+
+
+@when_any('java.ready', 'hadoop-plugin.java.ready')
+def set_java_home():
```

Comment:
Again, `@when_any` will never pass a param.  Are you sure you got all of the PRs and rebuilt every charm using them?  Something seems very off with this.